docs/crd/readme.md

```diff
@@ -0,0 +1,77 @@
+  optional string oauth_secret = 7; // The oauth secret for external users to use this deployment via an API.
+  map<string,string> annotations = 8; // Arbitrary annotations.
+}
+
+```
+
+For each predictor you should at a minimum specify:
+
+ * A unique name
+ * A PredictiveUnit graph that presents the tree of components to deploy.
+ * A componentSpec which describes the set of images for parts of your container graph that will be instigated as microservice containers. These containers will have been wrapped to work within the [internal API](../reference/internal-api.md). This component spec is a standard [PodTemplateSpec](https://kubernetes.io/docs/api-reference/extensions/v1beta1/definitions/#_v1_podtemplatespec).
+     * If you leave the ports empty for each container they will be added automatically and matched to the ports in the graph specification. If you decide to specify the ports manually they should match the port specified for the matching component in the grph specification.
+ * the number of replicas of this predictor to deploy
+
+```proto
+
+message PredictorSpec {
+  required string name = 1; // A unique name not used by any other predictor in the deployment.
+  required PredictiveUnit graph = 2; // A graph describing how the predictive units are connected together.
+  required k8s.io.api.core.v1.PodTemplateSpec componentSpec = 3; // A description of the set of containers used by the graph. One for each microservice defined in the graph.
+  optional int32 replicas = 4; // The number of replicas of the predictor to create.
+  map<string,string> annotations = 5; // Arbitrary annotations.
+}
+
+```
+
+The predictive unit graph is a tree. Each node is of a particular type with the leaf nodes being models. If the implementation is not specified then a microservice is assumed and you must define a matching named container within the componentSpec above. Each type of PredictiveUnit has a standard set of methods it is expected to manage, see [here](../reference/seldon-deployment.md). 
+
+For each node in the graph:
+
+ * A unique name. If the node describes a mircoservice then it must match a named container with the componentSpec
+ * The children nodes.
+ * The type of the predictive unit : MODEL, ROUTER, COMBINER etc
+ * The implementation. This can be left blank if it will be a microserice as this is the default otherwise choose from the avilable appropriate implementions provided internally.
+ * Methods. This can be left blank if you wish to follow the standard methods for your PredictiveNode type : see [here](../reference/seldon-deployment.md). 
+ * Endpoint. In here you should minimally if this a microservice specify whether the PredictiveUnit will use REST or gRPC. Ports will be defined automatically if not specified.
+ * Parameters. Specify any parameters you wish to pass to the PredictiveUnit. These will be passed as added environment variables as a JSON list of key:value pairs with the name PREDICTIVE_UNIT_PARAMETERS.
```

Comment:
Actually the json list is not of  key:value pairs, it has the format:

[
{
name:
value:
type:
},
... ]